### ISSUE TITLE
Support plain (non-SSL) connections

### ISSUE THREAD

**Issue by ronkorving** [indutny/spdy-transport#17]
As discussed on Twitter, this is a remaining TODO for this module.

Reference: https://httpwg.github.io/specs/rfc7540.html#discover-http

**Comment by ronkorving** [comment]
So, @indutny, you mentioned the challenge is to implement upgrade response/request. I'm confusing how HTTP/2 can even work without it, but can't find anything in the codebase to deal with upgrade requests. Does that mean that generally clients that support HTTP/2 will downgrade to HTTP/1.1 (instead of upgrading)?

Keep in mind, I'm new to HTTP/2. But am willing to give this a shot if nobody else will :)

**Comment by indutny** [comment]
@ronkorving it usually works on top of TLS, and there is ALPN TLS extension to negotiate the protocol ahead of time.

**Comment by ronkorving** [comment]
Got it, thanks. So it's for non-encrypted connections where UPGRADE becomes the method?

**Comment by indutny** [comment]
Yes, exactly!

**Comment by ronkorving** [comment]
Unfortunate that not a single browser seems to send `Upgrade: h2c`. Or am I doing something wrong here? I also tried returning an `Alt-Svc: h2c=":8080"` header as described [here](https://tools.ietf.org/html/draft-ietf-httpbis-alt-svc-06) to advise that I support HTTP2, but it doesn't seem to have any effect.

**Comment by indutny** [comment]
Indeed. I can't find any reference of `h2c` in the chromium's source code. This is very sad.

**Comment by ronkorving** [comment]
:( looks like for now I'm screwed then. My Raspberry Pi will either have to put up with encrypting streams of video, or I need to give up on HTTP2 (which I need, because I run into browser connection limits otherwise)... Sigh :(
